src/lib.js

```diff
@@ -222,8 +222,29 @@ var curlConverter = {
 
             this.headerPairs = {};
 
+            // if method is not given in the curl command
             if(!curlObj.request) {
-            	curlObj.request = "GET";
+            
+                // set method to POST if any of the param is given (-d, --data, --data-binary, --data-ascii)
+                if (curlObj.data.length > 0 || curlObj.dataAscii.length > 0 || curlObj.dataUrlencode.length > 0 || curlObj.dataBinary) {
+                    curlObj.request = "POST";
+                }
+                // set method to GET if there is -G or --get param in the curl command
+                else if (curlObj.get) {
```

Comment:
Are you sure of the precedence that you've coded? If I give both `--get` and `--data`, what'll be the method that's sent?

Reply:
According to curl specs, method will be GET in this case and it's working correctly.

Reply:
wait..how is it working? you're checking for `.data` first.